report_data.py

```diff
@@ -97,31 +156,170 @@ def read_raw_data(self):
+                        date=event.date,
+                        value=event.received_shares.total_buy_value()
+                        + event.withheld_shares.total_buy_value(),
+                    )
+                    bought_shares = awv.AWVEntryZ10RSUDeposit(
+                        date=event.date,
+                        quantity=event.received_shares.quantity,
+                        value=event.received_shares.total_buy_value()
+                        + event.withheld_shares.total_buy_value(),
+                    )
+                    withheld_shares = awv.AWVEntryZ10RSUTaxWithholding(
+                        date=event.date,
+                        quantity=event.withheld_shares.quantity,
+                        value=event.received_shares.total_buy_value(),
+                    )
+                    self.awv_z4_events.append(bonus)
+                    self.awv_z10_events.append(bought_shares)
+                    if event.withheld_shares.quantity > 0:
+                        self.awv_z10_events.append(withheld_shares)
+
+            elif isinstance(event, data_structures.ESPPEvent):
+                self.held_shares[event.symbol].push(event.received_shares)
+
+                if "NVDA" in event.symbol and event.currency == "USD":
+                    bonus = awv.AWVEntryZ4ESPPBonus(
+                        date=event.date,
+                        value=event.bonus,
+                    )
+
+                    bought_shares = awv.AWVEntryZ10ESPPDeposit(
+                        date=event.date,
+                        quantity=event.received_shares.quantity,
+                        value=event.received_shares.total_buy_value(),
+                    )
+
+                    self.awv_z4_events.append(bonus)
+                    self.awv_z10_events.append(bought_shares)
+
+            elif isinstance(event, data_structures.DividendEvent):
+                if event.currency != self.domestic_currency:
+                    self.held_forex[event.currency].push(event.received_net_dividend)
+
+                self.misc["Tax Withholding"].append(event.withheld_tax)
+                self.misc["Dividend Payments"].append(event.received_dividend)
+
+                # dividends should be small enough to not trigger AWV reportings
+
+            elif isinstance(event, data_structures.BuyEvent):
+                if event.currency != self.domestic_currency:
+                    # if not enough money, pop on FOREX Queue will fail
+                    tmp = self.held_forex[event.currency].pop(
+                        event.cost_of_shares,
+                        to_decimal(1),
+                        event.date,
+                    )
+                    self.sold_forex[event.currency].extend(tmp)
+
+                self.held_shares[event.symbol].push(event.received_shares)
+                self.misc["Fees"].append(event.paid_fees)
+
+                if event.currency == "USD" and "NVDA" in event.received_shares.symbol:
```

Comment:
Do we need special handling of NVDA USD?
The reporting is based on trading location, not currencies or tickers. So if I buy an asset in Luxembourg in EUR, that I also should declare.

Also, I myself keep my transactions based on ISIN numbers and I can imagine others can, too.

Reply:
For now, it was mainly about ease-of-use with data directly from Schwab. If you just have NVDA stocks to declare, these things can be automated without looking up exchange data, ISIN data, etc..
I'll add support for all symbols with placeholder values and remarks to fill them out for anything not corresponding to a NVDA-USD transaction.

Reply:
Yes, I think it's fine as a start and we can fine-tune it later if we want to.